src/value.cc

```diff
@@ -694,7 +694,7 @@ class bracket_visitor : public boost::static_visitor<Value>
     const auto i = convert_to_uint32(idx);
     if (i < str.size()) {
 			//Ensure character (not byte) index is inside the character/glyph array
-			if (i < g_utf8_strlen(str.c_str(), str.size()))	{
+			if ((int)i < g_utf8_strlen(str.c_str(), str.size()))	{
```

Comment:
`g_utf8_strlen` is guaranteed to return a positive integer, so rather wrap that return value in a `static_cast`.